src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt

```diff
@@ -3,40 +3,29 @@ package io.github.mojira.arisa.modules
 import arrow.core.Either
 import arrow.core.extensions.fx
 import io.github.mojira.arisa.domain.Issue
+import io.github.mojira.arisa.domain.Link
 import java.time.Instant
 
 class RemoveIdenticalLinkModule : Module {
-    data class GroupingLink(
-        val type: String,
-        val outwards: Boolean,
-        val key: String
-    )
-
     override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
         Either.fx {
             assertNotEmpty(links).bind()
 
             val removeLinkFunctions = links
-                .groupBy {
-                    GroupingLink(
-                        it.type,
-                        if (it.type.toLowerCase() == "relates") {
-                            true
-                        } else {
-                            it.outwards
-                        },
-                        it.issue.key
-                    )
-                }
-                .filterValues { it.size > 1 }
-                .values
-                .flatMap { list ->
-                    list.subList(1, list.size).map { it.remove }
+                .filter(::isRelatesLink)
+                .groupBy { it.issue.key }
+                .filterValues { it.size == 2 }
+                .map {
+                    // Always remove the outwards link belonging to the smaller ticket key,
+                    // so that when this module is triggered on both tickets, only one link is removed.
+                    it.value.find { l -> if (key < it.key) l.outwards else !l.outwards }?.remove
```

Comment:
I don't think this compares the ticket keys properly, as the key includes the letters and you're trying to compare them.

This doesn't always work properly, for example, "MC-14" is considered less than "MC-2".

I'd recomment comparing after splitting at the dash and converting the leftover string to an int.

Reply:
It really doesn't matter, we just need some consistent way to compare the two keys :eyes: